_meta/beat.yml

```diff
@@ -137,61 +183,7 @@ apm-server:
       # is changed, a matching index pattern needs to be specified here.
       #index_pattern: "apm-*-sourcemap*"
 
-
-  # If true (default), APM Server captures the IP of the instrumented service
-  # or the IP and User Agent of the real user (RUM requests).
-  #capture_personal_data: true
-
-  # Enable APM Server Golang expvar support (https://golang.org/pkg/expvar/).
-  #expvar:
-    #enabled: false
-
-    # Url to expose expvar.
-    #url: "/debug/vars"
-
-  # Instrumentation support for the server's HTTP endpoints and event publisher.
-  #instrumentation:
-    # Set to true to enable instrumentation of the APM Server itself.
-    #enabled: false
-
-    # Environment in which the APM Server is running on (eg: staging, production, etc.)
-    #environment: ""
-
-    # Remote hosts to report instrumentation results to.
-    #hosts:
-    #  - http://remote-apm-server:8200
-
-    # secret_token for the remote apm-servers.
-    #secret_token:
-
-  # A pipeline is a definition of processors applied to documents when ingesting them to Elasticsearch.
-  # Using pipelines involves two steps:
-  # (1) registering a pipeline
-  # (2) applying a pipeline during data ingestion (see `output.elasticsearch.pipeline`)
-  #
-  # You can manually register a pipeline, or use this configuration option to ensure
-  # the pipeline is loaded and registered at the configured Elasticsearch instances.
-  # Find the default pipeline configuration at `ingest/pipeline/definition.json`.
-  # Automatic pipeline registration requires the `output.elasticsearch` to be enabled and configured.
-  #register.ingest.pipeline:
-    # Registers APM pipeline definition in Elasticsearch on APM Server startup. Defaults to true.
-    #enabled: true
-    # Overwrites existing APM pipeline definition in Elasticsearch. Defaults to false.
-    #overwrite: false
-
-  # When ilm is set to `auto`, the APM Server checks a couple of preconditions:
-  # If a different output than Elasticsearch is configured, ILM will be disabled.
-  # If Elasticsearch output is configured, but specific `index` or `indices` settings are configured, ILM will be
-  # disabled, as it only works with default index settings.
-  # If the configured Elasticsearch instance is not eligible for ILM, ILM will also be disabled.
-  # If all preconditions are met, ILM will be enabled.
-  #
-  # When ILM is set to `true`, the APM Server ignores any configured index settings.
-  # For ILM to be applied, The configured output must be set to Elasticsearch and the instance
-  # needs to support ILM. Otherwise APM Server falls back to ordinary index management without ILM.
-  #
-  # Defaults to "auto". Disable ILM by setting it to `false`.
-  #ilm.enabled: "auto"
+  #---------------------------- APM Server - Agent Configuration ----------------------------
```

Comment:
Did you think how are we going to maintain these headers when configs can serve multiple purposes? eg, if we add another feature requiring Kibana, how this header is going to be named?

Reply:
We can simply rename the header if Kibana is used for something else in the future. Not sure I get your point.
With all the long configurations, I think it helps readability grouping them together.